test/e2e/nodepool_kv_advanced_multinet_test.go

```diff
@@ -0,0 +1,326 @@
+	if err := k.infra.CreateOVNKLayer2NAD("default"); err != nil {
+		return err
+	}
+
+	infraClient, err := k.infra.DiscoverClient()
+	if err != nil {
+		return err
+	}
+
+	t.Log("Creating a proxy dnsmasq pod")
+	dnsmasqPod := k.composeDNSMasqPod()
+	if err := infraClient.Create(k.infra.Ctx(), dnsmasqPod); err != nil {
+		return fmt.Errorf("failed creating dnsmasq pod: %w", err)
+	}
+
+	passthroughService := hcpmanifests.IngressDefaultIngressPassthroughService(k.infra.Namespace())
+	passthroughService.Name = fmt.Sprintf("%s-%s",
+		hcpmanifests.IngressDefaultIngressPassthroughServiceName,
+		k.infra.HostedCluster().Spec.Platform.Kubevirt.GenerateID)
+
+	t.Logf("Waiting for the default ingresss service to appear...")
+	findPassthroughServiceRetryConfig := wait.Backoff{
+		Steps:    15,
+		Duration: 5 * time.Second,
+	}
+	if err := retry.OnError(findPassthroughServiceRetryConfig, apierrors.IsNotFound, func() error {
+		return infraClient.Get(k.infra.Ctx(), client.ObjectKeyFromObject(passthroughService), passthroughService)
+	}); err != nil {
+		return err
+	}
+
+	t.Logf("Waiting for dnsmasq pod to have an address...")
+	err = wait.PollImmediate(5*time.Second, 2*time.Minute, func() (bool, error) {
+		if err := infraClient.Get(k.infra.Ctx(), client.ObjectKeyFromObject(dnsmasqPod), dnsmasqPod); err != nil {
+			return false, err
+		}
+		return dnsmasqPod.Status.PodIP != "", nil
+	},
+	)
+	if err != nil {
+		return err
+	}
+
+	ports := []discoveryv1.EndpointPort{}
+	for _, port := range passthroughService.Spec.Ports {
+		ports = append(ports, discoveryv1.EndpointPort{
+			Name:     &port.Name,
+			Protocol: &port.Protocol,
+			Port:     pointer.Int32(int32(port.TargetPort.IntValue())),
+		})
+	}
+	eps := discoveryv1.EndpointSlice{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: k.infra.Namespace(),
+			Name:      "test-e2e-nodepool-kv-advanced-multinet-default-ingress-passthrough-dnsmasq-ipv4",
+			Labels: map[string]string{
+				discoveryv1.LabelServiceName: passthroughService.Name,
+			},
+		},
+		AddressType: discoveryv1.AddressType(corev1.IPv4Protocol),
+		Ports:       ports,
+		Endpoints: []discoveryv1.Endpoint{{
+			Addresses: []string{dnsmasqPod.Status.PodIP},
+			Conditions: discoveryv1.EndpointConditions{
+				Ready:       pointer.Bool(true),
+				Serving:     pointer.Bool(true),
+				Terminating: pointer.Bool(false),
+			},
+		}},
+	}
+	t.Logf("Creating custom default ingress passthrough endpointslice")
+	if err := infraClient.Create(k.infra.Ctx(), &eps); err != nil {
+		return err
+	}
+
+	t.Logf("Waiting for vmi to report an address...")
+	vmiAddress := ""
+	findVMIAddressRetryConfig := wait.Backoff{
+		Steps:    120,
+		Duration: 2 * time.Second,
+	}
+	allErrors := func(error) bool { return true }
+	if err := retry.OnError(findVMIAddressRetryConfig, allErrors, func() error {
+		var err error
+		vmiAddress, err = k.firstVMIAddress()
+		if err != nil {
+			return err
+		}
+		return nil
+	}); err != nil {
+		return err
+	}
```

Comment:
Recent failure happened around here.

```
    nodepool_kv_advanced_multinet_test.go:191: Waiting for vmi to report an address...
    nodepool_test.go:236: 
        should succeed seting up the infra after creating the nodepool
        Expected success, but got an error:
            <*errors.errorString | 0xc002ee0340>: 
            missing interfaces addres at vmi
            {
                s: "missing interfaces addres at vmi",
            }
```

I think the issue here is probably that the retry loop should wait longer.

It can take an inconsistent amount of time for the VMIs to start after posting a new NodePool. the Disks have to be imported, which takes awhile. then there's also the issue with the default IP not consistently showing up on the VMI spec itself. The IP definitely eventually shows up, but it can flap back and forth on the VMI spec at times depending on if the VMI is rebooting (like after ignition is first applied)